src/node_util.cc

```diff
@@ -16,6 +16,22 @@ using v8::Proxy;
 using v8::String;
 using v8::Value;
 
+static void GetOwnNonIndicesProperties(
+    const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+  Local<Context> context = env->context();
+
+  v8::Local<v8::Object> object = args[0].As<v8::Object>();
+
+  // Return only non-enumerable properties by default.
+  v8::Local<v8::Array> properties = object
+    ->GetPropertyNames(context, v8::KeyCollectionMode::kOwnOnly,
+                       v8::ONLY_ENUMERABLE,
+                       v8::IndexFilter::kSkipIndices)
+      .ToLocalChecked();
```

Comment:
Same as above, does this ever fail? Instead, you could try something like

`if(!object->GetPropertyNames(...).ToLocal(&properties)) return;`

Reply:
I think you do, yes. The whole point of `ToLocalChecked` is that you must've checked the value earlier. You could either add a `CHECK` statement before this one, in which case it'll still crash (probably in a much better manner, but crash nonetheless) or use `ToLocal` as I did above.

Reply:
Since the original behavior of `Object.keys` is to throw if the keys cannot be collected, it makes more sense to return if the maybe is empty and throw in the JS land. (I guess this should fail if you throw an error in the `enumerate` proxy handler, though it is deprecated?)

Reply:
Yes, please use @ryzokuken’s suggestion here, I think this could otherwise fail when inspecting Proxies with an `ownKeys` handler. A `CHECK()` would have the same behaviour as `ToLocalChecked()`, so that’s not enough either.

Reply:
Thanks a lot! I learned something through your comments :+1:. It is now handled as suggested.